dev-packages/application-package/src/application-package.ts

```diff
@@ -29,11 +29,13 @@ export class ApplicationPackageOptions {
     readonly log?: ApplicationLog;
     readonly error?: ApplicationLog;
     readonly registry?: NpmRegistry;
+    readonly appTarget?: ApplicationProps.Target;
 }
 
 export type ApplicationModuleResolver = (modulePath: string) => string;
 
 export class ApplicationPackage {
+    private _appTarget?: ApplicationProps.Target;
```

Comment:
Why the underscore for the `_appTarget`? For clarification; for example, `_props` has the leading underscore, because it would be a name collision with the `props` property getter.

Alternatively, you can get rid of the `_appTarget` field, and do:
```ts
if (this.options.appTarget) {
  theia.target = this.options.appTarget;
}
```

Reply:
Oh you're right. I always forget about the auto prop assignments from args in Typescript. I'll get rid of the `_appTarget` prop and just reference `this.options.appTarget`.